src/topology/omega_complete_partial_order.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+-/
+import topology.basic
+import order.omega_complete_partial_order
+
+/-!
+# Scott Topological Spaces
+
+I don't know what they're good for but their notion
+of continuity is equivalent to continuity in ωCPOs.
+
+## Reference
+
+ * https://ncatlab.org/nlab/show/Scott+topology
+
+-/
+
+open omega_complete_partial_order
+open_locale classical
+
+universes u
+namespace scott_topological_space
+
+/--  -/
+def is_ωSup {α : Type u} [preorder α] (c : chain α) (x : α) : Prop :=
+(∀ i, c i ≤ x) ∧ (∀ y, (∀ i, c i ≤ y) → x ≤ y)
+
+variables (α : Type u) [preorder α]
+
+/-- The characteristic function of open sets is monotone and preserves
+the limits of chains. -/
+def is_open (s : set α) : Prop :=
+∃ (h : monotone s), ∀ (c : chain α) x, is_ωSup c x → is_ωSup (c.map ⟨s,h⟩) (s x)
```

Comment:
This is stated in a very weird way. It took me a while to figure out what `monotone s` is supposed to mean given that `s` is a set, not a function. I guess you are abusing the defeq here to mean "`s` is upward closed", but I would prefer if you said that directly. Also there is an occurence of `s x` in this definition which is also an abuse of set defeq.

Reply:
I agree that using `set` is confusing, but I like the spirit of this formulation. I think it would be even clearer to define an open set as being the inverse image of `True` by a function `\alpha -> Prop` that is continuous in the order-theoretic sense.

Reply:
What would be the relation between the function and the set? Also, how do you formulate the order theoretic continuity?

Reply:
I gather from the way this definition is stated that `set a` and `a -> Prop` are the same thing. Thus, the function I was talking about and the set would be the same thing. By "order-theoretic continuity", I meant that this function should satisfy the `omega_complete_partial_order.continuous'` predicate of the other file. (Note that `Prop` is a CPO for the order you are using here.)

Reply:
If we use `continuous'` though, we need `a` to be a CPO as well. Wouldn't that be sort of begging the question?